config/src/main/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.java

```diff
@@ -373,6 +375,32 @@ public AuthorizationManagerRequestMatcherRegistry anonymous() {
 			return access(AuthenticatedAuthorizationManager.anonymous());
 		}
 
+		public class AuthorizedUrlVariable {
+			private final String variable;
+
+			public AuthorizedUrlVariable(String variable) {
+				this.variable = variable;
+			}
+
+			public AuthorizationManagerRequestMatcherRegistry equalTo(Function<Authentication, String> function) {
+				return access((auth, requestContext) -> {
+					String value = requestContext.getVariables().get(variable);
+					return new AuthorizationDecision(function.apply(auth.get()).equals(value));
+				});
+			}
+
+			public AuthorizationManagerRequestMatcherRegistry equalTo(String expected) {
+				return access((auth, requestContext) -> {
+					String value = requestContext.getVariables().get(variable);
+					return new AuthorizationDecision(value.equals(expected));
+				});
+			}
+		}
```

Comment:
I don't think this has much value as it is. If I want the variable `{username}` to match `my-username`, why don't just use `.requestMatchers("/v2/user/my-username")`? For now, we can probably stick with just the `equalTo(Function<Authentication, String> function)` method. @jzheaux what do you think?

Reply:
Agreed. I think the confusion comes from [this line](https://github.com/spring-projects/spring-security/pull/13574/files#diff-36c789296b82aff789237431ca37c122773954f9f7c82846a5f5c6d90a3efa1cL1054) which is an overly complicated use of `WebExpressionAuthorizationManager`.

Reply:
Thank you for good opinion. I also agree with both of you and will delete that method!